crates/analyzer/src/traversal/expressions.rs

```diff
@@ -1726,184 +1800,186 @@ fn expr_call_type_attribute(
+                        &target_name
                     ),
-                    vec![Label::primary(
-                        target_span,
-                        format!(
-                            "expected a value of type `{}`, found type name",
-                            &class_name
-                        ),
-                    )],
-                    vec![],
-                )));
-            } else {
-                context.fancy_error(
-                    "Static functions need to be called with `::` not `.`",
-                    vec![Label::primary(
-                        field.span,
-                        "This is a static function (doesn't take a `self` parameter)",
-                    )],
-                    vec![format!(
-                        "Try `{}::{}(...)` instead",
-                        &class_name, &field.kind
-                    )],
-                );
-            }
+                )],
+                vec![],
+            )));
+        } else {
+            context.fancy_error(
+                "Static functions need to be called with `::` not `.`",
+                vec![Label::primary(
+                    field.span,
+                    "This is a static function (doesn't take a `self` parameter)",
+                )],
+                vec![format!(
+                    "Try `{}::{}(...)` instead",
+                    &target_name, &field.kind
+                )],
+            );
+        }
 
-            // Returns `true` if the current contract belongs to the same class as an input
-            // `callee_class`.
-            let is_same_class = |callee_class| match (context.root_item(), callee_class) {
-                (Item::Type(TypeDef::Contract(caller)), Class::Contract(callee)) => {
-                    caller == callee
-                }
-                (Item::Type(TypeDef::Struct(caller)), Class::Struct(callee)) => caller == callee,
-                _ => false,
-            };
+        // Returns `true` if the current contract belongs to the same class as an input
+        // `callee_class`.
+        let is_same_class = |callee_class| match (context.root_item(), callee_class) {
+            (Item::Type(TypeDef::Contract(caller)), Type::Contract(callee)) => caller == callee,
+            (Item::Type(TypeDef::Struct(caller)), Type::Struct(callee)) => caller == callee,
+            _ => false,
+        };
 
-            // Check for visibility of callee.
-            // Emits an error if callee is not public and caller doesn't belong to the same
-            // class as callee.
-            if !function.is_public(context.db()) && is_same_class(class) {
-                context.fancy_error(
+        // Check for visibility of callee.
+        // Emits an error if callee is not public and caller doesn't belong to the same
+        // class as callee.
+        if !sig.is_public(context.db()) && is_same_class(target_type.typ(context.db())) {
```

Comment:
Presumably this should be `!is_same_class(..)`? (I guess this is an uncaught mistake from a previous change)

Reply:
Nice catch. Unfortunately, just flipping the condition doesn't cut it here. I'm not sure if the code is exercised at all anymore. It inspired me to file https://github.com/ethereum/fe/issues/765

I will leave this as is to not bloat the PR but I'll investigate it on Monday (I'm a bit short on time today)

Reply:
Fixed in https://github.com/ethereum/fe/pull/767